includes/data-stores/class-wcs-customer-store-cached-cpt.php

```diff
@@ -213,7 +233,7 @@ public function set_empty_cache( $user_id ) {
 	 */
 	public function maybe_update_for_post_meta_change( $update_type, $subscription_id, $meta_key, $user_id, $old_user_id = '' ) {
 
-		if ( $this->get_meta_key() !== $meta_key ) {
+		if ( $this->get_meta_key() !== $meta_key && 'customer_id' !== $meta_key ) {
```

Comment:
I'm wondering if we should make changes so that calling `$this->get_meta_key()` returns the meta key we care about i.e. if HPOS is enabled then `$this->get_meta_key()` returns `'customer_id'`.

Not a big deal but I just thought of this when I saw this line.

To do this we'd have to look at changing `meta_key` to protected instead of private and then set it in the constructor

Reply:
I've updated this in [Add function to get the respective data key for the cache manager](https://github.com/Automattic/woocommerce-subscriptions-core/pull/304/commits/d1471f725a55bb029e1ca5c939176d8ec799a1f9). 

As I've mentioned, things are sort of in a halfway state in this class because of the CPT references and post references. There's more work to be done in this area for sure.